lib/matplotlib/collections.py

```diff
@@ -1094,15 +1098,22 @@ def set_verts(self, verts, closed=True):
             return
 
         # Fast path for arrays
-        if isinstance(verts, np.ndarray):
-            verts_pad = np.concatenate((verts, verts[:, :1]), axis=1)
+        if isinstance(verts, np.ndarray) and len(verts):
+            verts_pad = (np.concatenate((verts, verts[:, :1]), axis=1)
+                           .astype(mpath.Path.verts_type))
             # Creating the codes once is much faster than having Path do it
             # separately each time by passing closed=True.
             codes = np.empty(verts_pad.shape[1], dtype=mpath.Path.code_type)
             codes[:] = mpath.Path.LINETO
             codes[0] = mpath.Path.MOVETO
             codes[-1] = mpath.Path.CLOSEPOLY
-            self._paths = [mpath.Path(xy, codes) for xy in verts_pad]
+            example_path = mpath.Path(verts_pad[0], codes)
```

Comment:
```suggestion
            example_path = mpath.Path(verts_pad[0], closed=True)
            codes=example_path.codes
```
Instead of creating the codes manually above.

Reply:
Oh that's a nice one!